modules/visual_script/editor/visual_script_editor.cpp

```diff
@@ -1960,6 +1960,9 @@ void VisualScriptEditor::_generic_search(Vector2 pos, bool node_centered) {
 
 	new_connect_node_select->select_from_visual_script(script, false); // do not reset text
 
+#if 0
```

Comment:
Typically we delete code instead of defining them out.

Will need to check the visuals next after reviewing the code on the website.

Reply:
My intent was for the code owner to decide what to do about this code.  

I could not leave it, because it causes the popup to get pushed off screen.   However, I also could not fix it, because it isn't clear what it is trying to do.   There is a comment that it is trying to put the popup within the bounds of the graph.

I can't figure out why this would have been a requirement?   Surely if the main window is currently small on the screen, you don't want the popup to try to fit inside it?  It is a floating window.

I also could not determine intent from the code itself, because its calculation is so wrong.

I put the FIXME according to https://docs.godotengine.org/en/stable/community/contributing/code_style_guidelines.html#comment-style-guide but I couldn't leave the code enabled, because it messes up popup handling entirely.

I was hoping the code owner would know what the requirements/intent of this was and figure out what to do.   Is there some requirement that should be implemented here?   I didn't find anything in the guide about what to do in this scenario.

Reply:
So much for the background info. :)  I am of course happy to delete the code if you determine that is the right thing.

Reply:
> I could not leave it, because it causes the popup to get pushed off screen. However, I also could not fix it, because it isn't clear what it is trying to do. There is a comment that it is trying to put the popup within the bounds of the graph.
>
> I can't figure out why this would have been a requirement? Surely if the main window is currently small on the screen, you don't want the popup to try to fit inside it? It is a floating window.

You have to remember that Godot before 4.0 did not support multiple windows. So all dialogs were embedded in the main window (which can be reproduced in 4.0 by enabling single window mode).

So this code probably tries to make sure that the dialog in 3.x and in single window mode in 4.0 fits within the bounds of the graph editor. Since windowing changed in 4.0, this may no longer be needed, or may indeed be done wrong. So it can either be removed completely if the issue it tries to solve is no longer an issue, or fixed.

I'd suggest removing, after checking how things work in Single Window Mode.

Reply:
CC @swarnimarun who implemented that code in case you remember why it's needed and why it's done this way. But I strongly suspect the DisplayServer and multi window support changes in 4.0 pretty much invalidate what it was trying to do, so the simplest is probably to remove it if it causes issues, and make a new fix if there are still problems (e.g. in Single Window Mode).

Reply:
Great info, thanks.  Will do.

Reply:
@akien-mga was right.  This code doesn't do anything bad in ```--single-window```.   

Window::set_position(...) in single window mode takes coordinates relative to the main window, presumably like 3.x.

Window::set_position(...) in multi window takes absolute screen coordinates, so things can end up on an entirely different monitor for example.  I am going to start a separate discussion on that, but it is the reason why I initially said the calculation is wrong (for 4.x).

I tested it on both single and multi, and it is better to just remove the code and let the popup code position things, because it is single/multi aware.

Reply:
Sounds good.

Reply:
Removed.